ubuntu/16.04/usr/local/share/bootstrap/update_permissions.sh

```diff
@@ -17,12 +17,21 @@ function update_permissions() {
     REFERENCE="$1"
     read -r owner group owner_id group_id < <(stat -c '%U %G %u %g' "$REFERENCE")
     if [[ "$owner" = UNKNOWN ]]; then
-        APP_USER="$(randname)"
+        APP_USER="build"
+        if [ "$APP_USER_LOCAL_RANDOM" == 'true' ]; then
+            APP_USER="$(randname)"
+        fi
+        CODE_OWNER="$APP_USER"
         export APP_USER
+        export CODE_OWNER
+
         if [[ "$group" = UNKNOWN ]]; then
-            export APP_GROUP="$owner"
-            addgroup --system --gid "$group_id" "$APP_GROUP"
+            APP_GROUP="$owner"
+            CODE_GROUP="$APP_GROUP"
+            export APP_GROUP
+            export CODE_GROUP
+            groupadd --system --gid "$group_id" "$APP_GROUP" || groupmod -g "$group_id" "$APP_GROUP"
         fi
-        adduser --system --uid="$owner_id" --gid="$group_id" "$APP_USER"
+        useradd --create-home --system --uid "$owner_id" --gid "$group_id" "$APP_USER" || usermod -u "$owner_id" "$APP_USER"
```

Comment:
I'm wondering if changing the uid of the APP_USER will cause problems because it's home directory will no longer be readable